src/core/generators/query.ts

```diff
@@ -262,7 +262,7 @@ export const ${camel(`use-${name}`)} = <
     isRequestOptions,
     type,
   })}\n  ) => {
-  const queryKey = ${queryKeyFnName}(${properties});
+  const queryKey = options?.query?.queryKey ?? ${queryKeyFnName}(${properties});
```

Comment:
Can you move it below the `queryOptions` and do `queryOptions?.queryKey ?? ${queryKeyFnName}(${properties});`